src/DDTrace/Encoders/Json.php

```diff
@@ -12,10 +12,14 @@
      */
     public function encodeTraces(array $traces)
     {
-        return '[' . implode(',', array_map(function ($trace) {
-            return '[' . implode(',', array_map(function ($span) {
+        $notEmpty = function ($value) {
+            return !empty($value);
+        };
+
+        return '[' . implode(',', array_map(function ($trace) use ($notEmpty) {
+            return '[' . implode(',', array_filter(array_map(function ($span) {
                 return $this->encodeSpan($span);
-            }, $trace)) . ']';
+            }, $trace), $notEmpty)) . ']';
```

Comment:
When `array_filter`'s callback parameter is not supplied, entries equals to `false` (after casted to bool) will be removed. Which is the case here  with empty string

Reply:
Nice catch! Thank you.